### PR TITLE
Use docker-compose build/push and service push config

### PR DESCRIPTION
**Pull request by andytson-inviqa**
So that services can be added or removed easily

Due to intermediate image needs for build, this can't be easily resolved automatically without stronger language support (like php composer dependencies)

### CODE REVIEW

File: src/_base/harness/attributes/docker-base.yml [hunk 1]
```diff
@@ -26,17 +26,19 @@ attributes:
...
         VARNISH_HOSTNAME_TEMPLATE: "varnish-%d.varnish-headless"
     nginx:
+      publish: true
       ingress:
         annotations: {}
       environment:
         FPM_HOST: php-fpm
     console:
-      enabled: true
+      publish: true
```
Comment (andytson-inviqa) [inline src/_base/harness/attributes/docker-base.yml:36]:
```suggestion
      enabled: true
      publish: true
```

File: src/_base/harness/config/pipeline.yml [hunk 2]
```diff
@@ -5,49 +5,28 @@ command('app build'):
...
-
-command('app build tls-offload'): |
-  #!bash(harness:/docker/image/tls-offload)|@
-  passthru docker build -t @('docker.repository'):@('app.version')-tls-offload .
-
-command('app publish'):
+command('app build <service>'):
   env:
-    HAS_CRON: "= ('cron' in @('app.services') ? 'true' : 'false')"
+    SERVICE:    = input.argument('message')
```
Comment (andytson-inviqa) [inline src/_base/harness/config/pipeline.yml]:
```suggestion
    SERVICE:    = input.argument('service')
```

**Comment by kierenevans** [comment]
Does `docker-compose down --rmi local` clean up tagged images still? Seen a few things left around on Jenkins runners.

**Comment by kierenevans** [comment]
Never mind, we are doing this separately in destroy step: https://github.com/inviqa/harness-base-php/blob/0.9.x/src/_base/harness/scripts/destroy.sh#L13-L16

**Review by kierenevans — APPROVED** [review]

**Pull request merged**